parser/internal/parservX/parser_test.go

```diff
@@ -130,3 +150,110 @@ func assertParseFailure(t *testing.T, src string, message string, start int, end
 	assert.Equal(t, start, parseErr.Pos().Start)
 	assert.Equal(t, end, parseErr.Pos().End)
 }
+
+// Copy of testhelpers which allows functions to be passed to the parser
+var versionRegex = regexp.MustCompile(`//groq:version=(\S+)`)
+var paramRegex = regexp.MustCompile(`//groq:param:([^=]+)=([^\n]+)`)
+
+type Test struct {
+	Name      string
+	Query     string
+	Params    groq.Params
+	Functions map[ast.FunctionID]*ast.FunctionDefinition
+	Valid     bool
+	Version   *semver.Range
+	FileName  string
+}
+
+func ASTTest(
+	t *testing.T,
+	test *Test,
+	outputDir string,
+	parser func(query string, params groq.Params, functions map[ast.FunctionID]*ast.FunctionDefinition) (ast.Expression, error),
+) {
+	fileName := filepath.Join(outputDir, SnapshotFileName(test))
+
+	require.NoError(t, os.MkdirAll(filepath.Dir(fileName), 0755))
+
+	t.Logf("File: %s", fileName)
+
+	parsed, err := parser(test.Query, test.Params, test.Functions)
+	require.NoError(t, err)
+
+	type Result struct {
+		Query string
+		AST   ast.Expression
+	}
+
+	result := Result{
+		Query: test.Query,
+		AST:   parsed,
+	}
```

Comment:
Can we also compare the functions defined?
```suggestion
  type Result struct {
		Query     string
		AST       ast.Expression
		Functions map[ast.FunctionID]*ast.FunctionDefinition
	}

	result := Result{
		Query:     test.Query,
		AST:       parsed,
		Functions: test.Functions,
	}
```